### PR TITLE
More search filtering

### PR DESCRIPTION
**Pull request by stevage**
Surprisingly useful. Yes, we should have a proper UX with UI elements, but this was pretty easy to knock up.

### CODE REVIEW

File: lib/ViewModels/CatalogItemNameSearchProviderViewModel.js [hunk 1]
```diff
@@ -72,11 +104,52 @@ CatalogItemNameSearchProviderViewModel.prototype.search = function(searchText) {
...
+function itemMatchesFilters(item, searchFilters) {
+    if (searchFilters.typeIs.length > 0 && searchFilters.typeIs.indexOf(item.type.toLowerCase()) < 0) {
+        return false;
+    }
+    if (searchFilters.typeIsNot.length > 0 && searchFilters.typeIsNot.indexOf(item.type.toLowerCase()) >= 0) {
+        return false;
+    }
+    if (!item.url) {
+        // if no URL, it can't match any positive or negative filters.
+        return searchFilters.urlMatches.length >= 0;
```
Comment (kring) [inline lib/ViewModels/CatalogItemNameSearchProviderViewModel.js]: I think this logic is wrong.  It should return false when the item has no URL and there's a URL filter, not true, right?

**Comment by kring** [comment]
This looks good other than the one comment.

**Comment by stevage** [comment]
Thanks for catching that!

**Comment by kring** [comment]
There seems to be a problem with queries not containing a `url` portion.  For example, the query `url:gov ADSL` finds several catalog items, as expected.  But `ADSL` finds no results.

**Comment by kring** [comment]
@stevage for future reference, drop a note in here when you've committed some code that addresses review comments and it's ready for another review.  GitHub doesn't automatically notify of commits to pull requests, so a message is my reminder to come back and look again.

**Comment by stevage** [comment]
Ta, I was wondering about that.

**Pull request merged**